# Source: Andreas Gruenbacher agruen at suse.de 
# Desritiption: [Security] Long-standing xattr sharing bug
# Upstream: Included
# Revision Date: Tue, 09 Aug 2005 18:30:56 +0900
# Ref: http://acl.bestbits.at/pipermail/acl-devel/2005-February/001848.html
# Ref: http://www.novell.com/linux/security/advisories/2005_18_kernel.html
# Ref: http://lists.debian.org/debian-kernel/2005/08/msg00238.html
--- a/fs/ext2/xattr.c	2005-08-09 17:59:44.000000000 +0900
+++ b/fs/ext2/xattr.c	2005-08-09 17:59:46.000000000 +0900
@@ -977,6 +977,7 @@
 		if (IS_LAST_ENTRY(entry2))
 			return 1;
 		if (entry1->e_hash != entry2->e_hash ||
+		    entry1->e_name_index != entry2->e_name_index ||
 		    entry1->e_name_len != entry2->e_name_len ||
 		    entry1->e_value_size != entry2->e_value_size ||
 		    memcmp(entry1->e_name, entry2->e_name, entry1->e_name_len))
--- a/fs/ext3/xattr.c	2005-08-09 17:58:31.000000000 +0900
+++ b/fs/ext3/xattr.c	2005-08-09 17:58:50.000000000 +0900
@@ -1025,6 +1025,7 @@
 		if (IS_LAST_ENTRY(entry2))
 			return 1;
 		if (entry1->e_hash != entry2->e_hash ||
+		    entry1->e_name_index != entry2->e_name_index ||
 		    entry1->e_name_len != entry2->e_name_len ||
 		    entry1->e_value_size != entry2->e_value_size ||
 		    memcmp(entry1->e_name, entry2->e_name, entry1->e_name_len))
